channel/allocation.go

```diff
@@ -94,7 +96,7 @@ type (
 	// where all participants' assets are deposited.
 	// The same Asset should be shareable by multiple Allocation instances.
 	// Decoding happens with AppBackend.DecodeAsset.
-	Asset perunio.Encoder
+	Asset = encoding.BinaryMarshaler
```

Comment:
Use type def instead of type alias!?

https://yourbasic.org/golang/type-alias/

Reply:
It is changed to type def in the later commit (50fb4ee7b263fa55d9a3e12ef802505ee1f206f6) (where BinaryUnmarshaler was added), this is changed to type def. I will fix the older commits when rebasing/squashing. Else, it will break the current references.